### PR TITLE
Update Table.__getitem__, wrap StopIteration

### PR DESCRIPTION
**Pull request by mshriver**
return RowNotFound

### CODE REVIEW

File: src/widgetastic/widget/table.py [hunk 1]
```diff
@@ -456,7 +456,10 @@ def __getitem__(self, item):
         if self.table_tree:
             nodes = self.resolver.glob(self.table_tree, '/table/tbody/tr*')
             at_index = at_index + 1 if self._is_header_in_body else at_index or 1
-            return next(n.obj for n in nodes if n.position == at_index)
+            try:
+                return six.next(n.obj for n in nodes if n.obj.index == at_index)
+            except StopIteration:
+                raise RowNotFound('Row not found by index {} via {}'.format(at_index, item))
```
Comment (izapolsk) [inline src/widgetastic/widget/table.py]: :+1:

Comment (izapolsk) [inline src/widgetastic/widget/table.py]: :+1:

**Comment by izapolsk** [comment]
looks good but  wt.core and some other repos may expect StopIteration to be raised. So, RowNotFound may be not handled properly.

File: src/widgetastic/widget/table.py [hunk 2]
```diff
@@ -456,7 +456,10 @@ def __getitem__(self, item):
         if self.table_tree:
             nodes = self.resolver.glob(self.table_tree, '/table/tbody/tr*')
             at_index = at_index + 1 if self._is_header_in_body else at_index or 1
-            return next(n.obj for n in nodes if n.position == at_index)
+            try:
+                return six.next(n.obj for n in nodes if n.obj.index == at_index)
```
Comment (mshriver) [inline src/widgetastic/widget/table.py]: This change to the obj index fixed my test behavior, but isn't appropriate for a rowspan/colspan table.

My test is related to dynamic table, and detecting rowspan/colspan in the first place. Going to reduce this to changing from `StopIteration` to `RowNotFound` here

**Pull request merged**